src/map/battle.c

```diff
@@ -4979,7 +4986,10 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 			flag.cri = 1;
 	}
 	if (flag.cri) {
-		wd.type = BDT_CRIT;
+		if (battle_config.feature_enable_multi_crit == 1)
+			wd.type = (!(wd.type&BDT_MULTIHIT) ? BDT_CRIT : BDT_MULTICRIT);
```

Comment:
Why is this written like this, can't it just be?

```C
if (battle_config.feature_enable_multi_crit == 1 && (wd.type & BDT_MULTIHIT) != 0)
    wd.type = BDT_MULTICRIT;
```

Reply:
> Why is this written like this, can't it just be?
> 
> ```c
> if (battle_config.feature_enable_multi_crit == 1 && (wd.type & BDT_MULTIHIT) != 0)
>     wd.type = BDT_MULTICRIT;
> ```

Because if it is written like that there will be a random "1" damage showing if the critical attack didn't trigger the triple attack simultaneously.
![osiuwoM](https://github.com/HerculesWS/Hercules/assets/30449376/538a42dd-973d-4fb9-ae85-add851241386)

Reply:
That's unrelated, the code is just doing a useless check, plus as per my previous comment the use of a bitmask here is wrong as well

Reply:
Yes, I just found out that now, making a new commit.

Reply:
Done, please check the new commit 865838f